Compilation error when using memalign

Received compilation errors when running commands `make configure use=thread_sanitizer,pool_memalign` and ` make build`. Below is the error ouput. 
```
/ponyc/src/libponyrt/mem/pool_memalign.c:29:9: error: unused variable 'code' [-Werror,-Wunused-variable]
    int code = posix_memalign(&p, POOL_ALIGN, size);
        ^
/home/victor/Workspace/src/github.com/vijayee/ponyc/src/libponyrt/mem/pool_memalign.c:36:31: error: unused parameter 'index' [-Werror,-Wunused-parameter]
void ponyint_pool_free(size_t index, void* p)
                              ^
/home/victor/Workspace/src/github.com/vijayee/ponyc/src/libponyrt/mem/pool_memalign.c:46:7: error: unused variable 'code' [-Werror,-Wunused-variable]
  int code = posix_memalign(&p, POOL_ALIGN, size);
      ^
3 errors generated.
gmake[3]: *** [src/libponyrt/CMakeFiles/libponyrt.dir/build.make:584: src/libponyrt/CMakeFiles/libponyrt.dir/mem/pool_memalign.c.o] Error 1
gmake[3]: Leaving directory '/home/victor/Workspace/src/github.com/vijayee/ponyc/build/build_release'
gmake[2]: *** [CMakeFiles/Makefile2:395: src/libponyrt/CMakeFiles/libponyrt.dir/all] Error 2
gmake[2]: *** Waiting for unfinished jobs....

```

--- a/.github/workflows/pr.yml
+++ b/.github/workflows/pr.yml
@@ -411,9 +411,12 @@ jobs:
         include:
           - image: ghcr.io/ponylang/ponyc-ci-x86-64-unknown-linux-ubuntu22.04-builder:20230924
             debugger: lldb
-            directive: dtrace
+            directives: dtrace
+          - image: ghcr.io/ponylang/ponyc-ci-x86-64-unknown-linux-ubuntu22.04-builder:20230924
+            debugger: lldb
+            directives: pool_memalign
 
-    name: use ${{ matrix.directive }}
+    name: use ${{ matrix.directives }}
     container:
       image: ${{ matrix.image }}
       options: --user pony --cap-add=SYS_PTRACE --security-opt seccomp=unconfined
@@ -437,13 +440,13 @@ jobs:
           key: libs-${{ matrix.image }}-${{ hashFiles('Makefile', 'CMakeLists.txt', 'libs/CMakeLists.txt') }}
       - name: Build Debug Runtime
         run: |
-          make configure arch=x86-64 config=debug use=${{ matrix.directive }}
+          make configure arch=x86-64 config=debug use=${{ matrix.directives }}
           make build config=debug
       - name: Test with Debug Runtime
         run: make test-ci config=debug usedebugger='${{ matrix.debugger }}'
       - name: Build Release Runtime
         run: |
-          make configure arch=x86-64 config=release use=${{ matrix.directive }}
+          make configure arch=x86-64 config=release use=${{ matrix.directives }}
 
           make build config=release
       - name: Test with Release Runtime

--- a/.release-notes/fix-4431.md
+++ b/.release-notes/fix-4431.md
@@ -1,0 +1,3 @@
+## Fix compilation error when building with pool_memalign in release mode
+
+When attempting to build the Pony runtime with the `pool_memalign` option, users would encounter a compilation error if building a `release` rather than `debug` version of the runtime. We've fixed the compilation error and added CI testing to verify we don't get a regression.

--- a/src/libponyrt/mem/pool_memalign.c
+++ b/src/libponyrt/mem/pool_memalign.c
@@ -27,6 +27,7 @@ void* ponyint_pool_alloc(size_t index)
   else
   {
     int code = posix_memalign(&p, POOL_ALIGN, size);
+    (void)code;
     pony_assert(code == 0);
   }
 
@@ -35,6 +36,7 @@ void* ponyint_pool_alloc(size_t index)
 
 void ponyint_pool_free(size_t index, void* p)
 {
+  (void)index;
   pony_assert(index < POOL_COUNT);
 
   free(p);
@@ -44,6 +46,7 @@ static void* pool_alloc_size(size_t size)
 {
   void* p;
   int code = posix_memalign(&p, POOL_ALIGN, size);
+  (void)code;
   pony_assert(code == 0);
 
   return p;

